unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/NFTPromptHUD/NFTPromptHUDController.cs

```diff
@@ -1,43 +1,197 @@
 using DCL;
+using DCL.Helpers.NFT;
 using UnityEngine;
 
 public class NFTPromptHUDController : IHUD
 {
-    internal NFTPromptHUDView view { get; private set; }
+    internal const string VIEW_PREFAB_PATH = "NFTPromptHUD";
+
+    internal INFTPromptHUDView view { get; private set; }
+
+    private readonly OwnersInfoController ownersInfoController;
+
+    private Coroutine fetchNFTRoutine = null;
+    private NFTInfoSingleAsset? lastNFTInfo = null;
+
+    private bool isPointerInTooltipArea = false;
+    private bool isPointerInOwnerArea = false;
 
     public NFTPromptHUDController()
     {
-        view = UnityEngine.Object.Instantiate(Resources.Load<GameObject>("NFTPromptHUD")).GetComponent<NFTPromptHUDView>();
-        view.name = "_NFTPromptHUD";
-        view.content.SetActive(false);
+        view = Object.Instantiate(Resources.Load<GameObject>(VIEW_PREFAB_PATH))
+                     .GetComponent<NFTPromptHUDView>();
+        view.SetActive(false);
+
+        view.OnOwnerLabelPointerEnter += ShowOwnersTooltip;
+        view.OnOwnerLabelPointerExit += TryHideOwnersTooltip;
+        view.OnOwnersTooltipFocusLost += OnOwnersTooltipFocusLost;
+        view.OnOwnersTooltipFocus += OnOwnersTooltipFocus;
+        view.OnViewAllPressed += ShowOwnersPopup;
+        view.OnOwnersPopupClosed += HideOwnersPopup;
+
+        ownersInfoController = new OwnersInfoController(view.GetOwnerElementPrefab());
 
         if (Environment.i.world.sceneController != null)
             Environment.i.world.sceneController.OnOpenNFTDialogRequest += OpenNftInfoDialog;
```

Comment:
I know this has not been changed in this PR but we could decouple this from the sceneController by using our `DataStore`.

We could have a BaseVariable `currentNFTInDialog` of type:
```
public class NFTInfoModel
{
     public string assetContractAddress;
     public string tokenId;
     public string comment;
}
```

And make this HUD react to changes in that variable. Setting said variable to `null` would close the dialog and setting valid data would open it.

Therefore we could completly decouple the NFTPromptHUDController from any other system. It would also make cleaner how to trigger the HUD by just changing the model in the `DataStore`.

Reply:
Great suggestions.
I did all those changes if you want to take a look https://github.com/decentraland/unity-renderer/pull/539/commits/d42ded4ef0e066c14d94a87d3083fa127c34d4f9 before I merge tomorrow :)